File: OO/python/calculadora.py
#Senai Dendezeiros
#Nome: Matheus Cabral e Marcelo Ribeiro
#Calculadora orientada a objeto
#Para executar esse comando no script linux: php hello.py


class calculadora:

    resultado=0
    
    def __init__(self):
        print("Calculadora criada")

    def getResultado(self):
        print(self.resultado)
        return (self.resultado)

    def setResultado(self, result):
        self.resultado=result

    def somar(self, x,y):
        self.setResultado(x+y)
        return (self.getResultado())

    def subtrair(self, x,y):
        self.setResultado(x-y)
        return (self.getResultado())

    def multiplicar(self, x,y):
        self.setResultado(x*y)
        return (self.getResultado())

    def dividir(self, x,y):
        self.setResultado(x/y)
        return (self.getResultado())

File: OO/php/calculadora.php
<?php

    class Calculadora {

        public $resultado = 0;

        public function somar($x, $y){
            $this->resultado = ($x + $y);
            return $this->resultado;
        }

        public function subtrair($x, $y){
            $this->resultado = ($x - $y);
            return $this->resultado;
        }

        public function multiplicar($x, $y){
            $this->resultado = ($x * $y);
            return $this->resultado;
        }

        public function soma($x, $y){
            $this->resultado = ($x / $y);
            return $this->resultado;
        }
    }

    $resultado = new calculadora();

    print($resultado->somar(2,2));